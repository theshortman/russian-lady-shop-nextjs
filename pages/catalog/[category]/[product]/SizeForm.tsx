import { IProductItem } from "@/models/IProductItem";
import { FC } from "react";
import SizeList from "./SizeList";

interface SizeFormProps {
  sizes: IProductItem[];
}

const SizeForm: FC<SizeFormProps> = ({ sizes }) => {
  return (
    <form className="mb-3" action="" method="post">
      <div className="mb-2">
        <SizeList sizes={sizes} />
      </div>
      <div className="form-group">
        <button type="button" className="btn btn-danger text-white" disabled>
          ДОБАВИТЬ В КОРЗИНУ
        </button>
      </div>
    </form>
  );
};

export default SizeForm;
