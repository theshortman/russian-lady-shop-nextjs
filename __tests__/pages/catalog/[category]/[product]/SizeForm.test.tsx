import { render } from "@testing-library/react";
import SizeForm from "@/pages/catalog/[category]/[product]/SizeForm";

describe("Product page", () => {
  test("renders SizeForm unchanged", () => {
    const { container } = render(<SizeForm sizes={[]} />);
    expect(container).toMatchSnapshot();
  });
});
