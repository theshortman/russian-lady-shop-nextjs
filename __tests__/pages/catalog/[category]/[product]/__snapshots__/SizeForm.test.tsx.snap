// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page renders SizeForm unchanged 1`] = `
<div>
  <form
    action=""
    class="mb-3"
    method="post"
  >
    <div
      class="mb-2"
    />
    <div
      class="form-group"
    >
      <button
        class="btn btn-danger text-white"
        disabled=""
        type="button"
      >
        ДОБАВИТЬ В КОРЗИНУ
      </button>
    </div>
  </form>
</div>
`;
